src/QsCompiler/Core/SymbolTable/NamespaceManager.fs

```diff
@@ -1226,6 +1226,20 @@ type NamespaceManager
         finally
             syncRoot.ExitWriteLock()
 
+    /// Add an empty namespace named nsName if it does not already exist. Useful for adding the
+    /// stand-in namespace for notebooks.
+    member this.AddEmptyNamespaceIfNotExists(nsName) =
+        syncRoot.EnterWriteLock()
```

Comment:
I'm not sure how to write this without acquiring the write lock, I won't lie. (Like if it's possible to acquire the read lock and then only upgrade to the write lock if needed)

Reply:
Okay, it turns out that the code in TypeChecking can just call `CopyForExtension` so this isn't necessary. Will remove this in the next revision

Reply:
Removed